### DISCUSSION TITLE
linter / formatter / test runner

### DISCUSSION THREAD

**Discussion by arvigeus** [oven-sh/bun#712] [Ideas]
I love how Deno have these integrated, are there any plans for these down the road? Another plus for Deno integrated tools is that they are much faster than eslint / prettier (except test runner)

**Comment by Jarred-Sumner** [comment]
test runner: yes, `bun wiptest` is an early version. I think it will have a significant impact on the ecosystem once it graduates to `bun test`. Similar to what i expect from `bun install`, but perhaps a bigger impact for CI and less for individual developers.

formatter: maybe
linter: maybe

test runner matters more than both
I think formatter is lowest importance. Prettier is not fast, but it's rarely a bottleneck

**Reply by zigomir** [reply]
it’s imo not so much about the perf, it’s just so nice to ged rid of extra config files and modules to install for every project.

**Reply by The-Code-Monkey** [reply]
I think at the end of the day if we are going to be going down the road of running tests we need to also look at formatting because i think in a lot of people eyes, formatted code is just as useful as having code that works.

**Reply by mikefrancis** [reply]
@Jarred-Sumner Would it be possible to get an update on Bun's stance for 2025?

**Comment by ild0tt0re** [comment]
I think **[Oxc](https://github.com/web-infra-dev/oxc)** might be a good candidate to use in conjunction with **Bun**. It seems to have the same **focus on performance** that Bun has accustomed us to  🚀🚀 🚀

**Comment by birkskyum** [comment]
There is also [Biome](https://github.com/biomejs/biome) ( open continuation of Rome project ), which currently ( I hope it stay this way ) is a formatter/linter - coincidentally the only parts lacking in Bun :)
